addons/account/models/account_journal.py

```diff
@@ -604,6 +604,25 @@ def _alias_prepare_alias_name(self, alias_name, name, code, jtype, company):
                 alias_name = f"{alias_name}-{company_identifier}"
         return self.env['mail.alias']._sanitize_alias_name(alias_name)
 
+    @api.model
+    def _ensure_unique_alias_creation(self, vals, company):
+        """ Check uniqueness of the alias name within the given alias domain.
+        :param vals: the values of the journal.
+        :return: a unique alias name.
+        """
+        alias_name = vals.get('alias_name')
+        alias_domain_name = company.alias_domain_id.name
+
+        domain = [('alias_name', '=', alias_name)]
+        if alias_domain_name:
+            domain.append(('alias_domain', '=', alias_domain_name))
+        existing_alias = self.env['mail.alias'].search_count(domain, limit=1)
+
+        if existing_alias:
+            alias_name = f"{alias_name}-{vals.get('code', vals['name'][:5])}"
```

Comment:
Given the fact that the code is a required field, in which case could it not exist ?

Reply:
While the field is required, I wasn't entirely sure it's always in the dict, looking a bit above it seems the code may not exist in the case of imports (although it's filled here, maybe there are other cases?):
https://github.com/odoo/odoo/blob/5dafc19f72116f2784c0308b5df40f0ec3f117a9/addons/account/models/account_journal.py#L768

I wanted to play it safe because I'm not entirely familiar and I saw `get` being used just above:
https://github.com/odoo/odoo/blob/5dafc19f72116f2784c0308b5df40f0ec3f117a9/addons/account/models/account_journal.py#L779

WDYT ?

Reply:
Well I guess it doesn't hurt to keep the get, in case there is a scenario we don't think about.